primer/src/Primer/App.hs

```diff
@@ -983,11 +984,49 @@ applyProgAction prog = \case
                                 Just
                                   TypeDefConsFieldSelection
                                     { index
-                                    , meta = Right $ zt ^. _target % _typeMetaLens
+                                    , meta = Right $ Left $ zt ^. _target % _typeMetaLens
                                     }
                             }
                   }
           )
+  ParamKindAction tyName paramName id actions -> editModuleOfCrossType (Just tyName) prog $ \(mod, mods) defName def -> do
```

Comment:
> The current approach is not ideal since we traverse both branches even though only one will have a matching ID, and we don't fail if the ID isn't found anywhere

Do you think it is worth fixing this in this PR? I think it would be somewhat easy to do, but perhaps not worth it if you expect to revisit `Loc` soon.
Three ways spring to mind, all roughly implementing a `modifyKindNode :: ID -> Kind -> (Kind -> Kind) -> Either Error Kind` (or some variety of `focusKind`) function (where error could be "id not found" or "multiple id"):
- just do it in the "findNode" style -- we may already have the ability to use a generic uniplate zipper on things which `HasID` to find a particular node
- since kinds are simple, we could use some uniplate magic: something like `contexts` to get a focus on each node, then `filter` this by ID, then do the transform
- some small applicative-state hack on this current code where we work not in the current monad `m`, but in something like `StateT Int m` (or maybe `AccumT` or `WriterT` may be better) where every time the `modifyKind` actually runs the function, it also emits a `+1`. We then check that we counted to exactly `1` (i.e. did exactly one edit). (It may be nicer to work in the "0-1-many" monoid, rather than the integers, as that more clearly expresses what we care about).

> We also end up performing a full typechecking pass

This seems fine to me (at least, it is consistent with other actions)